File: Projects/ABASTECIMIENTOS/ABASTECIMIENTOS_ENTIDADES/Entidades/CONTRATOS.vb
''' -----------------------------------------------------------------------------
''' Project	 : ABASTECIMIENTOS_EL
''' Class	 : EL.CONTRATOS
''' 
''' -----------------------------------------------------------------------------
''' <summary>
''' Clase que permite tener un registro de la tabla ABASTECIMIENTOSCONTRATOS en memoria
''' </summary>
''' <remarks>
''' Generado con GenApp V1.6.2.0, Carías y Asociados, (info@cariasyasociados.com)
''' </remarks>
''' <history>
''' 	[GenApp]	22/02/2007	Created
''' </history>
''' -----------------------------------------------------------------------------
Public Class CONTRATOS
    Inherits entidadBase

#Region " Propiedades "

    Private _IDESTABLECIMIENTO As Int32
    Public Property IDESTABLECIMIENTO() As Int32
        Get
            Return _IDESTABLECIMIENTO
        End Get
        Set(ByVal value As Int32)
            _IDESTABLECIMIENTO = value
        End Set
    End Property

    Private _IDPROVEEDOR As Int32
    Public Property IDPROVEEDOR() As Int32
        Get
            Return _IDPROVEEDOR
        End Get
        Set(ByVal value As Int32)
            _IDPROVEEDOR = value
        End Set
    End Property

    Private _NOMBREPROVEEDOR As String
    Public Property NOMBREPROVEEDOR() As String
        Get
            Return _NOMBREPROVEEDOR
        End Get
        Set(ByVal value As String)
            _NOMBREPROVEEDOR = value
        End Set
    End Property

    Private _IDCONTRATO As Int64
    Public Property IDCONTRATO() As Int64
        Get
            Return _IDCONTRATO
        End Get
        Set(ByVal value As Int64)
            _IDCONTRATO = value
        End Set
    End Property

    Private _NUMEROCONTRATO As String
    Public Property NUMEROCONTRATO() As String
        Get
            Return _NUMEROCONTRATO
        End Get
        Set(ByVal value As String)
            _NUMEROCONTRATO = value
        End Set
    End Property

    Private _IDTIPODOCUMENTO As Int16
    Public Property IDTIPODOCUMENTO() As Int16
        Get
            Return _IDTIPODOCUMENTO
        End Get
        Set(ByVal value As Int16)
            _IDTIPODOCUMENTO = value
        End Set
    End Property

    Private _IDPLANTILLA As Int32
    Public Property IDPLANTILLA() As Int32
        Get
            Return _IDPLANTILLA
        End Get
        Set(ByVal value As Int32)
            _IDPLANTILLA = value
        End Set
    End Property

    Private _TIPOPERSONA As String
    Public Property TIPOPERSONA() As String
        Get
            Return _TIPOPERSONA
        End Get
        Set(ByVal value As String)
            _TIPOPERSONA = value
        End Set
    End Property

    Private _REPRESENTANTELEGAL As String
    Public Property REPRESENTANTELEGAL() As String
        Get
            Return _REPRESENTANTELEGAL
        End Get
        Set(ByVal value As String)
            _REPRESENTANTELEGAL = value
        End Set
    End Property

    Private _PERSONERIAJURIDICA As String
    Public Property PERSONERIAJURIDICA() As String
        Get
            Return _PERSONERIAJURIDICA
        End Get
        Set(ByVal value As String)
            _PERSONERIAJURIDICA = value
        End Set
    End Property

    Private _FECHAGENERACION As DateTime
    Public Property FECHAGENERACION() As DateTime
        Get
            Return _FECHAGENERACION
        End Get
        Set(ByVal value As DateTime)
            _FECHAGENERACION = value
        End Set
    End Property

    Private _FECHAAPROBACION As DateTime
    Public Property FECHAAPROBACION() As DateTime
        Get
            Return _FECHAAPROBACION
        End Get
        Set(ByVal value As DateTime)
            _FECHAAPROBACION = value
        End Set
    End Property

    Private _FECHADISTRIBUCION As DateTime
    Public Property FECHADISTRIBUCION() As DateTime
        Get
            Return _FECHADISTRIBUCION
        End Get
        Set(ByVal value As DateTime)
            _FECHADISTRIBUCION = value
        End Set
    End Property

    Private _CODIGOLICITACION As String
    Public Property CODIGOLICITACION() As String
        Get
            Return _CODIGOLICITACION
        End Get
        Set(ByVal value As String)
            _CODIGOLICITACION = value
        End Set
    End Property

    Private _FECHAINICIOENTREGA As DateTime
    Public Property FECHAINICIOENTREGA() As DateTime
        Get
            Return _FECHAINICIOENTREGA
        End Get
        Set(ByVal value As DateTime)
            _FECHAINICIOENTREGA = value
        End Set
    End Property

    Private _IDESTADOCONTRATO As Int16
    Public Property IDESTADOCONTRATO() As Int16
        Get
            Return _IDESTADOCONTRATO
        End Get
        Set(ByVal value As Int16)
            _IDESTADOCONTRATO = value
        End Set
    End Property

    Private _IDCALIFICACIONCUMPLIMIENTO As Int16
    Public Property IDCALIFICACIONCUMPLIMIENTO() As Int16
        Get
            Return _IDCALIFICACIONCUMPLIMIENTO
        End Get
        Set(ByVal value As Int16)
            _IDCALIFICACIONCUMPLIMIENTO = value
        End Set
    End Property

    Private _IDCALIFICACIONCALIDAD As Int16
    Public Property IDCALIFICACIONCALIDAD() As Int16
        Get
            Return _IDCALIFICACIONCALIDAD
        End Get
        Set(ByVal value As Int16)
            _IDCALIFICACIONCALIDAD = value
        End Set
    End Property

    Private _IDMODALIDADCOMPRA As Byte
    Public Property IDMODALIDADCOMPRA() As Byte
        Get
            Return _IDMODALIDADCOMPRA
        End Get
        Set(ByVal value As Byte)
            _IDMODALIDADCOMPRA = value
        End Set
    End Property

    Private _DESCRIPCIONMODALIDADCOMPRA As String
    Public Property DESCRIPCIONMODALIDADCOMPRA() As String
        Get
            Return _DESCRIPCIONMODALIDADCOMPRA
        End Get
        Set(ByVal value As String)
            _DESCRIPCIONMODALIDADCOMPRA = value
        End Set
    End Property

    Private _NUMEROMODALIDADCOMPRA As String
    Public Property NUMEROMODALIDADCOMPRA() As String
        Get
            Return _NUMEROMODALIDADCOMPRA
        End Get
        Set(ByVal value As String)
            _NUMEROMODALIDADCOMPRA = value
        End Set
    End Property

    Private _FUENTESFINANCIAMIENTO As String
    Public Property FUENTESFINANCIAMIENTO() As String
        Get
            Return _FUENTESFINANCIAMIENTO
        End Get
        Set(ByVal value As String)
            _FUENTESFINANCIAMIENTO = value
        End Set
    End Property

    Private _RESPONSABLEDISTRIBUCION As String
    Public Property RESPONSABLEDISTRIBUCION() As String
        Get
            Return _RESPONSABLEDISTRIBUCION
        End Get
        Set(ByVal value As String)
            _RESPONSABLEDISTRIBUCION = value
        End Set
    End Property

    Private _RESOLUCIONADJUDICACION As String
    Public Property RESOLUCIONADJUDICACION() As String
        Get
            Return _RESOLUCIONADJUDICACION
        End Get
        Set(ByVal value As String)
            _RESOLUCIONADJUDICACION = value
        End Set
    End Property

    Private _MODIFICATIVASCONTRATO As String
    Public Property MODIFICATIVASCONTRATO() As String
        Get
            Return _MODIFICATIVASCONTRATO
        End Get
        Set(ByVal value As String)
            _MODIFICATIVASCONTRATO = value
        End Set
    End Property

    Private _TIPODOCUMENTO As String
    Public Property TIPODOCUMENTO() As String
        Get
            Return _TIPODOCUMENTO
        End Get
        Set(ByVal value As String)
            _TIPODOCUMENTO = value
        End Set
    End Property

    Private _MONTOCONTRATO As Decimal
    Public Property MONTOCONTRATO() As Decimal
        Get
            Return _MONTOCONTRATO
        End Get
        Set(ByVal value As Decimal)
            _MONTOCONTRATO = value
        End Set
    End Property

    Private _AUUSUARIOCREACION As String
    Public Property AUUSUARIOCREACION() As String
        Get
            Return _AUUSUARIOCREACION
        End Get
        Set(ByVal value As String)
            _AUUSUARIOCREACION = value
        End Set
    End Property

    Private _AUFECHACREACION As DateTime
    Public Property AUFECHACREACION() As DateTime
        Get
            Return _AUFECHACREACION
        End Get
        Set(ByVal value As DateTime)
            _AUFECHACREACION = value
        End Set
    End Property

    Private _AUUSUARIOMODIFICACION As String
    Public Property AUUSUARIOMODIFICACION() As String
        Get
            Return _AUUSUARIOMODIFICACION
        End Get
        Set(ByVal value As String)
            _AUUSUARIOMODIFICACION = value
        End Set
    End Property

    Private _AUFECHAMODIFICACION As DateTime
    Public Property AUFECHAMODIFICACION() As DateTime
        Get
            Return _AUFECHAMODIFICACION
        End Get
        Set(ByVal value As DateTime)
            _AUFECHAMODIFICACION = value
        End Set
    End Property

    Private _ESTASINCRONIZADA As Int16
    Public Property ESTASINCRONIZADA() As Int16
        Get
            Return _ESTASINCRONIZADA
        End Get
        Set(ByVal value As Int16)
            _ESTASINCRONIZADA = value
        End Set
    End Property

    Private _ACTANOTARIAL As String
    Public Property ACTANOTARIAL() As String
        Get
            Return _ACTANOTARIAL
        End Get
        Set(ByVal value As String)
            _ACTANOTARIAL = value
        End Set
    End Property

    Private _RESOLUCION As String
    Public Property RESOLUCION() As String
        Get
            Return _RESOLUCION
        End Get
        Set(ByVal value As String)
            _RESOLUCION = value
        End Set
    End Property

    Private _MODIFICATIVA As String
    Public Property MODIFICATIVA() As String
        Get
            Return _MODIFICATIVA
        End Get
        Set(ByVal value As String)
            _MODIFICATIVA = value
        End Set
    End Property

#End Region

End Class

File: Projects/SINAB_Entidades/Helpers/UACIHelpers/Contratos.cs
using System;
using System.Collections;
using System.Collections.Generic;
using System.Linq;
using System.Security.Cryptography;
using Microsoft.CSharp;
using SINAB_Entidades.Tipos;

namespace SINAB_Entidades.Helpers.UACIHelpers
{
    public static class Contratos
    {
        public static List<SAB_UACI_CONTRATOS> ObtenerTodos(int idEstablecimiento, int idEstadoContrato)
        {
            var db = new SinabEntities();

            return db.SAB_UACI_CONTRATOS.
                Where(c =>
                      c.IDESTABLECIMIENTO == idEstablecimiento &&
                      c.IDESTADOCONTRATO == idEstadoContrato).OrderByDescending(c => c.FECHAGENERACION).ToList();
        }
        public static List<SAB_UACI_CONTRATOS> ObtenerTodos(int idEstablecimiento, int idEstadoContrato, string usuarioCreacion)
        {
            var db = new SinabEntities();

            return db.SAB_UACI_CONTRATOS.
                Where(c =>
                      c.IDESTABLECIMIENTO == idEstablecimiento &&
                      c.AUUSUARIOCREACION.Contains(usuarioCreacion) &&
                      c.IDESTADOCONTRATO == idEstadoContrato
                      ).OrderByDescending(c => c.FECHAGENERACION).ToList();
        }


        public static List<BaseContratos> ObtenerTodos(decimal idEstablecimiento, decimal idAlmacen, decimal idTipoDocumento, decimal idModalidad, decimal idProveedor, int entrega, string producto)
        {
            using (var db = new SinabEntities())
            {
                return ObtenerTodos(db, idEstablecimiento, idAlmacen, idTipoDocumento, idModalidad, idProveedor, entrega,
                    producto);

            }
        }
        public static List<BaseContratos> ObtenerTodos(SinabEntities db, decimal idEstablecimiento, decimal idAlmacen, decimal idTipoDocumento, decimal idModalidad, decimal idProveedor, int entrega, string producto)
        {

            var res = (from c in db.SAB_UACI_CONTRATOS
                       join aec in db.SAB_UACI_ALMACENESENTREGACONTRATOS on new
                       {
                           c.IDESTABLECIMIENTO,
                           c.IDPROVEEDOR,
                           c.IDCONTRATO
                       } equals new
                       {
                           aec.IDESTABLECIMIENTO,
                           aec.IDPROVEEDOR,
                           aec.IDCONTRATO
                       }

                       join pc in db.SAB_UACI_PRODUCTOSCONTRATO on new
                       {
                           aec.IDESTABLECIMIENTO,
                           c.IDPROVEEDOR,
                           c.IDCONTRATO,
                           aec.SAB_UACI_ENTREGACONTRATO.RENGLON
                       } equals new
                       {
                           pc.IDESTABLECIMIENTO,
                           pc.IDPROVEEDOR,
                           pc.IDCONTRATO,
                           pc.RENGLON
                       }

                       join cp in db.vv_CATALOGOPRODUCTOS on pc.IDPRODUCTO equals cp.IDPRODUCTO
                       where
                           (c.IDESTABLECIMIENTO == idEstablecimiento || idEstablecimiento == 0) &&
                           (c.IDTIPODOCUMENTO == idTipoDocumento || idTipoDocumento == 0) &&
                           (c.IDMODALIDADCOMPRA == idModalidad || idModalidad == 0) &&
                           (c.IDPROVEEDOR == idProveedor || idProveedor == 0) &&
                           (aec.IDALMACENENTREGA == idAlmacen || idAlmacen == 0) &&
                           (cp.CORRPRODUCTO == producto || producto == "")
                       group aec by new
                       {
                           Suministro = cp.DESCSUMINISTRO,
                           contrato = c
                       }
                           into s

                           let ct = s.Key.contrato
                           select new BaseContratos()
                           {
                               IDESTABLECIMIENTO = ct.IDESTABLECIMIENTO,
                               IDPROVEEDOR = ct.IDPROVEEDOR,
                               IDCONTRATO = ct.IDCONTRATO,
                               TipoDocumento = ct.SAB_CAT_TIPODOCUMENTOCONTRATO.DESCRIPCION,
                               NumeroDocumento = ct.NUMEROCONTRATO,
                               FechaDocumento = ct.FECHADISTRIBUCION,
                               TipoCompra = ct.SAB_CAT_MODALIDADESCOMPRA.DESCRIPCION,
                               NumeroCompra = ct.NUMEROMODALIDADCOMPRA,
                               Proveedor = ct.SAB_CAT_PROVEEDORES.NOMBRE,
                               MONTOCONTRATO = ct.MONTOCONTRATO,
                               CantidadEntregada = s.Sum(c => c.CANTIDADENTREGADA),
                               Cantidad = s.Sum(c => c.CANTIDAD),
                               Suministros = (from pct in ct.SAB_UACI_PRODUCTOSCONTRATO
                                              join vv in db.vv_CATALOGOPRODUCTOS on pct.IDPRODUCTO
                                                  equals vv.IDPRODUCTO
                                              group pct by new { vv.CORRSUMINISTRO, vv.DESCSUMINISTRO }
                                                  into gr
                                                  select gr.Key.DESCSUMINISTRO),
                               FuentesFinanciamiento =
                                       ct.SAB_UACI_FUENTEFINANCIAMIENTOSCONTRATOS.Select(
                                           ffc => ffc.SAB_CAT_FUENTEFINANCIAMIENTOS.NOMBRE)

                           });

            switch (entrega)
            {
                case 1:
                    res = res.Where(c => c.CantidadEntregada == 0);
                    break;
                case 2:
                    res = res.Where(c => c.CantidadEntregada < c.Cantidad);
                    break;
                case 3:
                    res = res.Where(c => c.CantidadEntregada == c.Cantidad);
                    break;
            }
            var list = res.ToList().OrderByDescending(c => c.FechaDocumento).ToList();


            return list;


        }

        #region OPERACIONES SOBRE CONTRATO
        public static SAB_UACI_CONTRATOS Obtener(SAB_UACI_CONTRATOS contrato)
        {
            using (var db = new SinabEntities())
            {
                return Obtener(contrato, db);
            }
        }

        public static void Guardar(SAB_UACI_CONTRATOS contrato)
        {
            using (var db = new SinabEntities())
            {
                if (contrato.IDCONTRATO == 0)
                {
                    var recs = db.SAB_UACI_CONTRATOS.Where(c =>
                                                           c.IDESTABLECIMIENTO == contrato.IDESTABLECIMIENTO &&
                                                           c.IDPROVEEDOR == contrato.IDPROVEEDOR);
                    long lastId = 1;
                    if (recs.Any()) lastId = recs.Max(c => c.IDCONTRATO) + 1;
                    contrato.IDCONTRATO = lastId;

                    db.SAB_UACI_CONTRATOS.AddObject(contrato);
                }
                else
                {
                    var rec =
                        Obtener(
                            new SAB_UACI_CONTRATOS()
                            {
                                IDCONTRATO = contrato.IDCONTRATO,
                                IDESTABLECIMIENTO = contrato.IDESTABLECIMIENTO,
                                IDPROVEEDOR = contrato.IDPROVEEDOR
                            }, db);
                    if (rec != null)
                    {
                        rec.NUMEROCONTRATO = contrato.NUMEROCONTRATO;
                        rec.IDTIPODOCUMENTO = contrato.IDTIPODOCUMENTO;
                        rec.IDPLANTILLA = contrato.IDPLANTILLA;
                        rec.TIPOPERSONA = contrato.TIPOPERSONA;
                        rec.REPRESENTANTELEGAL = contrato.REPRESENTANTELEGAL;
                        rec.PERSONERIAJURIDICA = contrato.PERSONERIAJURIDICA;
                        rec.FECHAGENERACION = contrato.FECHAGENERACION;
                        rec.FECHAAPROBACION = contrato.FECHAAPROBACION;
                        rec.FECHADISTRIBUCION = contrato.FECHADISTRIBUCION;
                        rec.CODIGOLICITACION = contrato.CODIGOLICITACION;
                        rec.FECHAINICIOENTREGA = contrato.FECHAINICIOENTREGA;
                        rec.IDESTADOCONTRATO = contrato.IDESTADOCONTRATO;
                        rec.IDCALIFICACIONCUMPLIMIENTO = contrato.IDCALIFICACIONCUMPLIMIENTO;
                        rec.IDCALIFICACIONCALIDAD = contrato.IDCALIFICACIONCALIDAD;
                        rec.IDMODALIDADCOMPRA = contrato.IDMODALIDADCOMPRA;
                        rec.NUMEROMODALIDADCOMPRA = contrato.NUMEROMODALIDADCOMPRA;
                        rec.MONTOCONTRATO = contrato.MONTOCONTRATO;
                        rec.AUUSUARIOCREACION = contrato.AUUSUARIOCREACION;
                        rec.AUFECHACREACION = contrato.AUFECHACREACION;
                        rec.AUUSUARIOMODIFICACION = contrato.AUUSUARIOMODIFICACION;
                        rec.AUFECHAMODIFICACION = contrato.AUFECHAMODIFICACION;
                        rec.ESTASINCRONIZADA = contrato.ESTASINCRONIZADA;
                        rec.ACTANOTARIAL = contrato.ACTANOTARIAL;
                        rec.RESOLUCION = contrato.RESOLUCION;
                        rec.MODIFICATIVA = contrato.MODIFICATIVA;
                    }
                }
                db.SaveChanges();
            }
        }

        public static SAB_UACI_CONTRATOS Guardar(SAB_UACI_CONTRATOS contrato, SinabEntities db)
        {
            if (contrato.IDCONTRATO == 0)
            {
                var recs = db.SAB_UACI_CONTRATOS.Where(c =>
                                                       c.IDESTABLECIMIENTO == contrato.IDESTABLECIMIENTO &&
                                                       c.IDPROVEEDOR == contrato.IDPROVEEDOR);
                long lastId = 1;
                if (recs.Any()) lastId = recs.Max(c => c.IDCONTRATO) + 1;
                contrato.IDCONTRATO = lastId;

                db.SAB_UACI_CONTRATOS.AddObject(contrato);
                db.SaveChanges();
                return contrato;
            }

            var rec =
                Obtener(
                    new SAB_UACI_CONTRATOS()
                    {
                        IDCONTRATO = contrato.IDCONTRATO,
                        IDESTABLECIMIENTO = contrato.IDESTABLECIMIENTO,
                        IDPROVEEDOR = contrato.IDPROVEEDOR
                    }, db);
            if (rec != null)
            {
                rec.NUMEROCONTRATO = contrato.NUMEROCONTRATO;
                rec.IDTIPODOCUMENTO = contrato.IDTIPODOCUMENTO;
                rec.IDPLANTILLA = contrato.IDPLANTILLA;
                rec.TIPOPERSONA = contrato.TIPOPERSONA;
                rec.REPRESENTANTELEGAL = contrato.REPRESENTANTELEGAL;
                rec.PERSONERIAJURIDICA = contrato.PERSONERIAJURIDICA;
                rec.FECHAGENERACION = contrato.FECHAGENERACION;
                rec.FECHAAPROBACION = contrato.FECHAAPROBACION;
                rec.FECHADISTRIBUCION = contrato.FECHADISTRIBUCION;
                rec.CODIGOLICITACION = contrato.CODIGOLICITACION;
                rec.FECHAINICIOENTREGA = contrato.FECHAINICIOENTREGA;
                rec.IDESTADOCONTRATO = contrato.IDESTADOCONTRATO;
                rec.IDCALIFICACIONCUMPLIMIENTO = contrato.IDCALIFICACIONCUMPLIMIENTO;
                rec.IDCALIFICACIONCALIDAD = contrato.IDCALIFICACIONCALIDAD;
                rec.IDMODALIDADCOMPRA = contrato.IDMODALIDADCOMPRA;
                rec.NUMEROMODALIDADCOMPRA = contrato.NUMEROMODALIDADCOMPRA;
                rec.MONTOCONTRATO = contrato.MONTOCONTRATO;
                rec.AUUSUARIOCREACION = contrato.AUUSUARIOCREACION;
                rec.AUFECHACREACION = contrato.AUFECHACREACION;
                rec.AUUSUARIOMODIFICACION = contrato.AUUSUARIOMODIFICACION;
                rec.AUFECHAMODIFICACION = contrato.AUFECHAMODIFICACION;
                rec.ESTASINCRONIZADA = contrato.ESTASINCRONIZADA;
                rec.ACTANOTARIAL = contrato.ACTANOTARIAL;
                rec.RESOLUCION = contrato.RESOLUCION;
                rec.MODIFICATIVA = contrato.MODIFICATIVA;
            }

            db.SaveChanges();
            return rec;
        }

        public static void Borrar(SAB_UACI_CONTRATOS contrato)
        {
            //asegurando RollBack
            using (var db = new SinabEntities())
            {
                var rec = Obtener(contrato, db);


                if (rec == null) return;
                //Verificar antes de borrar
                if (rec.SAB_ALM_RECIBOSRECEPCION.Any())
                    throw new Exception("Contrato no puede ser borrado despues de recibido");

                //Borra los responsables
                foreach (var rdc in rec.SAB_UACI_RESPONSABLEDISTRIBUCIONCONTRATO.ToList())
                {
                    db.SAB_UACI_RESPONSABLEDISTRIBUCIONCONTRATO.DeleteObject(rdc);
                }

                //borra las fuentes de financiamiento
                foreach (var ffc in rec.SAB_UACI_FUENTEFINANCIAMIENTOSCONTRATOS.ToList())
                {
                    db.SAB_UACI_FUENTEFINANCIAMIENTOSCONTRATOS.DeleteObject(ffc);
                }

                //Borrar productos contrato
                foreach (var ec in rec.SAB_UACI_PRODUCTOSCONTRATO.SelectMany(pc => pc.SAB_UACI_ENTREGACONTRATO).ToList())
                {
                    db.SAB_UACI_ENTREGACONTRATO.DeleteObject(ec);
                }

                foreach (var pc in rec.SAB_UACI_PRODUCTOSCONTRATO.ToList())
                {
                    db.SAB_UACI_PRODUCTOSCONTRATO.DeleteObject(pc);
                }

                //borra el contrato
                db.SAB_UACI_CONTRATOS.DeleteObject(rec);
                db.SaveChanges();

            }
        }

        public static SAB_UACI_CONTRATOS Obtener(SAB_UACI_CONTRATOS contrato, SinabEntities db)
        {
            var rec = db.SAB_UACI_CONTRATOS.FirstOrDefault(c =>
                                                           c.IDCONTRATO == contrato.IDCONTRATO &&
                                                           c.IDESTABLECIMIENTO == contrato.IDESTABLECIMIENTO &&
                                                           c.IDPROVEEDOR == contrato.IDPROVEEDOR);
            return rec;
        }

        public static SAB_UACI_CONTRATOS Obtener(int idContrato, int idEstablecimiento, int idProveedor, SinabEntities db)
        {
            var rec = db.SAB_UACI_CONTRATOS.FirstOrDefault(c =>
                                                           c.IDCONTRATO == idContrato &&
                                                           c.IDESTABLECIMIENTO == idEstablecimiento &&
                                                           c.IDPROVEEDOR == idProveedor);
            return rec;
        }
        #endregion OPERACIONES SOBRE CONTRATO
    }
}
